include/xframe/xaxis_default.hpp

```diff
@@ -0,0 +1,326 @@
+/***************************************************************************
+* Copyright (c) 2017, Johan Mabille, Sylvain Corlay, Wolf Vollprecht and   *
+* Martin Renou                                                             *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XFRAME_XAXIS_DEFAULT_HPP
+#define XFRAME_XAXIS_DEFAULT_HPP
+
+#include <utility>
+#include <vector>
+#include <ostream>
+#include <iterator>
+
+#include "xtl/xiterator_base.hpp"
+
+#include "xaxis_base.hpp"
+#include "xaxis.hpp"
+
+namespace xf
+{
+    template <class L, class T>
+    class xaxis_default_iterator;
+
+    /*****************
+     * xaxis_default *
+     *****************/
+
+    template <class L, class T>
+    class xaxis_default : public xaxis_base<xaxis_default<L, T>>
+    {
+    public:
+
+        using base_type = xaxis_base<xaxis_default>;
+        using self_type = xaxis_default<L, T>;
+        using axis_type = xaxis<L, T>;
+        using key_type = typename base_type::key_type;
+        using label_list = std::vector<key_type>;
+        using mapped_type = typename base_type::mapped_type;
```

Comment:
You should add the definitions of `value_type`, `reference`, `const_reference`, `pointer` and `const_pointer` here.